File: packages/parser-go-strings-comments/fixtures/comments-and-strings.go
// Package greeting builds friendly messages for the CLI.
package greeting

/* approximate */
const pi = 3.14

/**
 * Adds two numbers together.
 */
func Add(a, b int) int {
	return a + b
}

/* Combines a name with a fixed prefix. */
func Format(name string) string {
	prefix := "see http://example.com" // not a real comment: "/* nope */"
	newline := '\n'
	_ = newline
	initial := 'A'
	_ = initial
	quoteInside := "she said \"hi\" then left"
	_ = quoteInside
	pattern := `C:\path\to\file "quoted" // not a comment`
	_ = pattern
	return prefix + name
}

File: packages/parser-rust-strings-comments/fixtures/comments-and-strings.rs
//! Module-level inner doc comment.

/// Adds two numbers together.
pub fn add(a: i32, b: i32) -> i32 {
    // running total
    a + b
}

/* approximate */
pub fn approx() -> f64 {
    3.14
}

/**
 * Computes the square of a number.
 */
pub fn square(x: i32) -> i32 {
    x * x
}

/*!
 * Inner doc block comment describing this section.
 */
pub fn documented() {}

pub fn greeting() -> &'static str {
    "see http://example.com"
}

pub fn escaped() -> &'static str {
    "she said \"hi\" then left"
}

pub fn initial() -> char {
    'A'
}

pub fn magic_bytes() -> &'static [u8] {
    b"binary payload marker"
}
